src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,100 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int DOUBLE_CAPACITY = 2;
+    private static final int NEGATIVE_COEFFICIENT = -1;
+    private int size;
+    private final float threshold;
+    private Node<K, V>[] hashTable;
+
+    public MyHashMap() {
+        hashTable = new Node[DEFAULT_CAPACITY];
+        threshold = hashTable.length * LOAD_FACTOR;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> currentNote = new Node<>(key, value, null);
+        resizeArray();
+        int index = valueIndex(keyHash(key));
+        if (hashTable[index] != null) {
+            Node<K, V> newNode = hashTable[index];
+            while (newNode != null) {
+                if ((newNode.key == key)
+                        || (key != null && key.equals(newNode.key))) {
+                    newNode.value = value;
+                    return;
+                }
+                if (newNode.next == null) {
+                    newNode.next = currentNote;
+                    size++;
+                    return;
+                }
+                newNode = newNode.next;
+            }
+        }
+        hashTable[index] = currentNote;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = valueIndex(keyHash(key));
+        Node<K, V> thisNode = hashTable[index];
+        while (hashTable[index] != null) {
```

Comment:
Why do we check **hashTable[index]** in condition? In my opinion, it's not correct. Maybe better to check **thisNode**?